### PR TITLE
AppChrome: Make side pane feature more generic and state owned by AppChromeService

### PR DESCRIPTION
**Pull request by torkelo**
https://github.com/grafana/grafana/pull/91892 runs into a messy issue of AppChrome having to depend on state inside the scope dashboards list component (if it's open or not). 

Not sure I like the fact that we always render ScopedDashboards on all pages (and that AppChrome has a dependency on it). Feels the wrong way around. esp since it's only shown on dashboard pages currently. currently it's rendered and activated on all pages even dashboards fetched no matter the page (if browser was closed with it opened )

This explores adding a concept of a side pane to AppChrome that can be added/rendered same way pages render the nav toolbar actions (via component that updates AppChromeService state). this is not ideal, best would be if this was all rendered inside the Page.

### CODE REVIEW

File: public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx [hunk 1]
```diff
@@ -61,6 +63,7 @@ export function DashboardSceneRenderer({ model }: SceneComponentProps<DashboardS
       {!editPanel && (
         <div className={cx(styles.pageContainer, hasControls && styles.pageContainerWithControls)}>
           <NavToolbarActions dashboard={model} />
+          <ScopesDashboards />
```
Comment (bfmatei) [inline public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx:65]: The suggested dashboards list should appear on all pages, not only when viewing a dashboard. The idea behind it is to allow quick navigation to a dashboard from other areas in Grafana.

This is an example of a workflow:
- Navigate to homepage
- Pick a scope
- Expand the suggested dashboards list and navigate to one of the dashboards
- Navigate away to another area in Grafana where scopes make sense (for the time being there ain't one, but just conceptually)
- Do some work in that area, even changing the scope(s)
- Navigate to one of the suggested dashboards

File: public/app/core/components/AppChrome/AppChrome.tsx [hunk 2]
```diff
@@ -152,6 +164,17 @@ const getStyles = (theme: GrafanaTheme2, searchBarHidden: boolean) => {
...
+        ? {
+            position: 'fixed',
+            height: `calc(100% - ${searchBarHidden ? TOP_BAR_LEVEL_HEIGHT : TOP_BAR_LEVEL_HEIGHT * 2}px)`,
+            zIndex: 2,
+          }
+        : {
+            zIndex: theme.zIndex.navbarFixed,
+          }
+    ),
+    sidePaneLeft: css(
```
Comment (ashharrison90) [inline public/app/core/components/AppChrome/AppChrome.tsx:177]: maybe we should set `width: MENU_WIDTH` here?

File: public/app/core/components/AppChrome/AppChrome.tsx [hunk 3]
```diff
@@ -188,7 +214,10 @@ const getStyles = (theme: GrafanaTheme2, searchBarHidden: boolean) => {
       }
     ),
     pageContainerMenuDocked: css({
-      paddingLeft: '300px',
+      paddingLeft: MENU_WIDTH,
+    }),
+    pageContainerMenuDockedScopes: css({
```
Comment (ashharrison90) [inline public/app/core/components/AppChrome/AppChrome.tsx:219]: we should remove mention of `Scopes` to keep this generic

File: public/app/core/components/AppChrome/AppChromeSidePane.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  // We use useLayoutEffect here to make sure that the chrome is updated before the page is rendered
+  // This prevents flickering actions when going from one dashboard to another for example
+  useLayoutEffect(() => {
+    chrome.update({ sidePaneLeft: left });
+  });
+  return null;
+});
+
+AppChromeSidePane.displayName = 'AppChromeSidePane';
```
Comment (ashharrison90) [inline public/app/core/components/AppChrome/AppChromeSidePane.tsx:24]: can we just use `<AppChromeUpdate sidePaneLeft={<Component />} />` directly, or do you think having this separate wrapper is cleaner? 🤔 (i could go either way tbh... right now it feels like a bit of extra boilerplate, but could be useful in the future)

**Review by ashharrison90 — COMMENTED** [review]
nice! 🙇

File: public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx:65]: @bfmatei currently the scope selector and dashboard list is only rendered on the dashboard page, is this just a current temp thing?

Comment (bfmatei) [reply public/app/features/dashboard-scene/scene/DashboardSceneRenderer.tsx:65]: Yeah. On the long run we want to expand the usage to other areas such as plugins, explore etc.

File: public/app/core/components/AppChrome/AppChromeSidePane.tsx (hunk 4 above)
Comment (torkelo) [reply public/app/core/components/AppChrome/AppChromeSidePane.tsx:24]: @ashharrison90 we could, I tried it but they override each other the way they currently works

File: public/app/core/components/AppChrome/AppChrome.tsx (hunk 2 above)
Comment (torkelo) [reply public/app/core/components/AppChrome/AppChrome.tsx:177]: yea, ideally the white styling and border should be set here as well

**Pull request closed without merging**